### PR TITLE
Adding cluster and data array factories

### PR DESCRIPTION
**Pull request by devonbush**
This is the factories from the raw-counts-ingest branch, and then I added two new ones for cluster and data_array.  These are going to be handy for testing the new visualization/explore endpoints

### CODE REVIEW

**Review by bistline — APPROVED** [review]
Really like where this is heading.  Once I'm finished with the `jb-code-coverage` PR, we should investigate where else we can integrate this functionality to make tests faster and more reliable.

File: test/api/site/explore_test.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require 'api_test_helper'
+
+class SearchControllerTest < ActionDispatch::IntegrationTest
+  include Devise::Test::IntegrationHelpers
+  include Requests::JsonHelpers
+  include Requests::HttpHelpers
+
+  HOMO_SAPIENS_FILTER = { id: 'NCBITaxon_9606', name: 'Homo sapiens' }
+  NO_DISEASE_FILTER = { id: 'MONDO_0000001', name: 'disease or disorder' }
```
Comment (jlchang) [inline test/api/site/explore_test.rb:9]: Can we talk through the intent of NO_DISEASE_FILTER? I suspect its inclusion in this test is related to its function in search_controller_test.rb and I want to make sure that what we're doing is still appropriate.

File: test/factories/cluster_group.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    study { study_file.study }
+    # create a study but mark as detached, so a Terra workspace is not created
+    factory :cluster_group_with_cells do
+      transient do
+        # cell_data is a hash of three (or 4) arrays: cells, x and y and z
+        cell_data {
+          {
+            x: [1, 2, 3],
+            y: [1, 2, 3],
+            text: ['cellA', 'cellB', 'cellC']
```
Comment (eweitz) [inline test/factories/cluster_group.rb:17]: `cells` seems like the intended field here.

`text` is similar, but as seen [in `load_cluster_group_data_array_points`](https://github.com/broadinstitute/single_cell_portal_core/blob/4661694410edc44c99af0ffde77d153b9a0195b7/app/controllers/site_controller.rb#L1562), it's an array comprised of cell names and other text. `render_cluster` responses reveal it's used only for tooltips in Plotly.  It's also the likeliest candidate for removal from our server-side-constructed Plotly object -- as it's the largest element in the cluster response, and ought to be computable client-side at runtime.  So relying on `cells` would also be more robust to performance-related refactoring.

File: test/factories/cluster_group.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+FactoryBot.define do
+  # gets a study object, defaulting to the first user found.  Auto-appends a unique number to the name, and a note
+  # in the description of the study, to aid in test DB uniqueness and cleanup efforts
+  factory :cluster_group do
+    cluster_type { '2d' }
```
Comment (eweitz) [inline test/factories/cluster_group.rb:4]: Given `{name: :z, type: 'coordinates'}` below, it seems `cluster_type` should be parameterized.

File: test/api/site/explore_test.rb (hunk 1 above)
Comment (devonbush) [reply test/api/site/explore_test.rb]: Ooops, this was a copy-paste error -- this doesn't belong here.  I just copied the test structure of this test from search_controller (as you guessed).  I'll delete

File: test/api/site/explore_test.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @basic_study_cluster_file = FactoryBot.create(:study_file,
+                                                  name: 'clusterA.txt',
+                                                  file_type: 'Cluster',
+                                                  study: @basic_study)
+    @cluster_group = FactoryBot.create(:cluster_group_with_cells,
+                                       study_file: @basic_study_cluster_file,
+                                       cell_data: {
+                                         x: [1, 4 ,6],
+                                         y: [7, 5, 3],
+                                         text: ['A', 'B', 'C']
```
Comment (eweitz) [inline test/api/site/explore_test.rb:28]: See comment beginning "`cells` seems like the intended field here".

**Review by eweitz — CHANGES_REQUESTED** [review]
This looks useful!

After renaming a field as described below, I think this will be ready.  I also suggested a parameterization, but I could also see deferring that as an enhancement.

File: test/factories/cluster_group.rb (hunk 3 above)
Comment (eweitz) [reply test/factories/cluster_group.rb:4]: Thanks for correcting my misunderstanding -- this already is parameter, and '2d' is the default.  What you have here seems ideal.

**Review by eweitz — APPROVED** [review]

**Comment by eweitz** [comment]
For auditability, please append a related Jira ticket in the title.

**Pull request merged**